src/invoice/store.rs

```diff
@@ -20,4 +31,105 @@ pub trait InvoiceStore {
         payment_hash: Hash256,
         preimage: Hash256,
     ) -> Result<(), InvoiceError>;
+    // A payment to an invoice is made by sending a TLC over some channels
+    // (possibly multiple when atomic multi-path payment support is out).
+    // This function returns all the channels that were used to pay an invoice.
+    fn get_invoice_channel_info(&self, payment_hash: &Hash256) -> Vec<InvoiceChannelInfo>;
+    // This function is used to add a channel (with the amount paid through this channel)
+    // to the list of channels that were used to pay an invoice.
+    fn add_invoice_channel_info(
+        &self,
+        payment_hash: &Hash256,
+        invoice_channel_info: InvoiceChannelInfo,
+    ) -> Result<Vec<InvoiceChannelInfo>, InvoiceError>;
+}
+
+#[derive(Copy, Clone, Serialize, Deserialize)]
+pub struct InvoiceChannelInfo {
+    pub channel_id: Hash256,
+    pub amount: u128,
+}
+
+impl InvoiceChannelInfo {
+    pub fn new(channel_id: Hash256, amount: u128) -> Self {
+        Self { channel_id, amount }
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum SettleInvoiceError {
+    #[error("Invoice not found")]
+    InvoiceNotFound,
+    #[error("Hash mismatch")]
+    HashMismatch,
+    #[error("Internal error: {0}")]
+    InternalError(String),
+}
+
+// This function is used to settle an invoice. It is extracted from the `InvoiceRpcServerImpl` struct
+// to share it with test functions.
+pub(crate) fn settle_invoice<S: InvoiceStore>(
```

Comment:
I don't think extract function is necessary, it introduces unnecessary ChannelCommand/NetworkActorCommand/NetworkActorMessage dependencies to the store module.